### PR TITLE
Linux101: filesystem

### PR DESCRIPTION
**Pull request by core-man**
Related to #67.

**Linux filesystem:**
- `/`
- `/home`

**File operation:**
- check files and directories
- make files and directories
- copy files and directories
- move files and directories
- remove files and directories
- link files and directories

**File path:**
- relative path: `.`, `..`
- absolute path: `/home/seismo-learn`

**File permission:**
- `rwx` 
- `chmod`
- `sudo`

**References:**
- Filesystem Hierarchy: https://101.lug.ustc.edu.cn/Ch05/#fhs
- File operation
  - http://www.1zlab.com/wiki/Hello-Ubuntu/operate-file-1/
  - http://www.1zlab.com/wiki/Hello-Ubuntu/operate-file-2/
  - https://101.lug.ustc.edu.cn/Ch03/#operate-files-and-dirs
  - https://xzchsia.github.io/2020/03/05/linux-hard-soft-link/
- File path: http://www.1zlab.com/wiki/Hello-Ubuntu/operate-file-1/#3-cd-
- File permission
  - https://101.lug.ustc.edu.cn/Ch05/#file-permission
  - https://101.lug.ustc.edu.cn/Ch05/#users-and-groups

### CODE REVIEW

File: source/computer/linux101.rst [hunk 1]
```diff
@@ -49,15 +49,236 @@ Linux 入门
 熟悉 Linux 文件系统
 ^^^^^^^^^^^^^^^^^^^
 
+Linux 文件系统就像一颗树一样，从 :file:`/` 目录开始，这个特殊的目录称为根目录。
+一般情况下，根目录下一般有 :file:`/home`\ 、\ :file:`/root`\ 、\ :file:`/bin`\ 、
+:file:`/usr`\ 、\ :file:`/lib`\ 、\ :file:`/opt` 等目录。根目录下的这些目录的具体含义
+由文件系统层次结构标准决定，大部分 Linux 发行版遵循此标准，或由此标准做了细小的调整。例如，
```
Comment (seisman) [inline source/computer/linux101.rst:55]:
```suggestion
:file:`/usr`\ 、\ :file:`/lib`\ 、\ :file:`/opt` 等目录。
```

File: source/computer/linux101.rst [hunk 2]
```diff
@@ -49,15 +49,236 @@ Linux 入门
...
+    $ cd ~
+    # 使用 mkdir（make directory，即创建目录）命令新建以下目录
+    $ mkdir codes software workspace
+    # 在 workspace 目录中新建 source 目录
+    $ mkdir workspace/source
+
+    # 进入 workspace 目录
+    $ cd workspace
+    # 使用 touch 命令创建新文件 hello-world.md 和 seiso-learn.md
+    $ touch hello-world.md seiso-learn.md
```
Comment (seisman) [inline source/computer/linux101.rst:115]: Add a `ls` command here to check these files.

File: source/computer/linux101.rst [hunk 3]
```diff
@@ -49,15 +49,236 @@ Linux 入门
...
+
+    # 进入 ~/workspace 目录
+    $ cd ~/workspace
+
+复制文件和目录::
+
+    # 使用 cp（copy，即复制）命令复制 hello-world.md 文件到同一目录下，并重命名为 hello-world-cp.md
+    $ cp hello-world.md hello-world-cp.md
+    # 复制 hello-world.md 和 seiso-learn.md 文件到 source 目录下
+    $ cp hello-world.md seiso-learn.md source
```
Comment (seisman) [inline source/computer/linux101.rst:127]: Add `ls` commands to check the results of each operation.

Same to all other commands.

Comment (core-man) [reply source/computer/linux101.rst]: done in ee9cc03.

File: source/computer/linux101.rst (hunk 2 above)
Comment (core-man) [reply source/computer/linux101.rst:122]: done in ee9cc03.

File: source/computer/linux101.rst [hunk 4]
```diff
@@ -49,15 +49,317 @@ Linux 入门
 熟悉 Linux 文件系统
 ^^^^^^^^^^^^^^^^^^^
 
+Linux 文件系统就像一颗树一样，从 :file:`/` 目录开始，这个特殊的目录称为根目录。
+一般情况下，根目录下一般有 :file:`/home`\ 、\ :file:`/root`\ 、\ :file:`/bin`\ 、
+:file:`/usr`\ 、\ :file:`/lib`\ 、\ :file:`/opt` 等目录。
```
Comment (seisman) [inline source/computer/linux101.rst:54]:
```suggestion
根目录下一般有 :file:`/home`\ 、\ :file:`/root`\ 、\ :file:`/bin`\ 、
:file:`/usr`\ 、\ :file:`/lib`\ 、\ :file:`/opt` 等目录。
```

File: source/computer/linux101.rst [hunk 5]
```diff
@@ -49,15 +49,317 @@ Linux 入门
...
+-  :file:`/opt` 目录：某些闭源或商业软件（如 Matlab、Google Chrome、Google Earth）
+   默认将软件安装到此目录下
+
+日常科研中，我们大多数情况下只在 :file:`/home` 目录下工作。假设用户名是 seismo-learn，
+该用户的家目录便是 :file:`/home/seismo-learn`。在 Linux 系统下，可用 :file:`~` 代表
+家目录。Linux 系统安装后，自动创建的家目录下的目录有 :file:`~/Desktop`\ 、\ :file:`~/Downloads`\ 、
+:file:`~/Documents` 等。熟悉 Linux 系统后可以参考\
+:doc:`《文件管理实践经验》<best-practices/file-organization>`\ 和\
+:doc:`《软件安装实践经验》<best-practices/software-installation>`\
+进一步组织与管理家目录，以提高工作效率。
```
Comment (seisman) [inline source/computer/linux101.rst:68]: We should mention that the home directory is `/Users/XXX` on macOS.

File: source/computer/linux101.rst [hunk 6]
```diff
@@ -49,15 +49,317 @@ Linux 入门
...
+    # 查看根目录中的子目录和文件
+    $ ls
+    bin   etc   lib    mnt  proc  run   srv  tmp  var
+    boot  dev   home  lib64  media   opt  root  sbin  sys  usr
+    # 切换回家目录（以下任一操作均可）
+    $ cd /home/seismo-learn
+    $ cd ~
+    $ cd
+
+    # 检查当前所在目录
```
Comment (seisman) [inline source/computer/linux101.rst:102]:
```suggestion
    # 确认已经切换回家目录下
```

File: source/computer/linux101.rst [hunk 7]
```diff
@@ -49,15 +49,317 @@ Linux 入门
...
+    # 进入家目录
+    $ cd ~
+    # 使用 mkdir（make directory，即创建目录）命令新建以下目录
+    $ mkdir codes software workspace
+    $ ls
+    codes  Desktop  Documents  Downloads  software  workspace
+    # 在 workspace 目录中新建 source 目录
+    $ mkdir workspace/source
+    $ ls workspace/source
+    source
```
Comment (seisman) [inline source/computer/linux101.rst:117]:
```suggestion
    $ ls workspace/
    source
```

File: source/computer/linux101.rst [hunk 8]
```diff
@@ -49,15 +49,317 @@ Linux 入门
...
+
+    # 进入 workspace 目录
+    $ cd workspace
+    # 使用 touch 命令创建新文件 hello-world.md 和 seiso-learn.md
+    $ touch hello-world.md seiso-learn.md
```
Comment (seisman) [inline source/computer/linux101.rst:122]: Please change these files to `.txt`, which users are more familiar with.

Comment (core-man) [reply source/computer/linux101.rst]: Done in 145bb17.

File: source/computer/linux101.rst (hunk 5 above)
Comment (core-man) [reply source/computer/linux101.rst]: Done in 6eacc91.

File: source/computer/linux101.rst [hunk 9]
```diff
@@ -49,15 +49,317 @@ Linux 入门
...
+    $ ls
+    hello-world-hard.txt  hello-world.txt  hello-world-soft.txt  source
+    # 删除 hello-world.txt 文件的软链接
+    $ rm hello-world-soft.txt
+    $ ls
+    hello-world-hard.txt  hello-world.txt  source
+    # 删除 hello-world.txt 文件的硬链接
+    $ rm hello-world-hard.txt
+    $ ls
+    hello-world.txt  source
```
Comment (seisman) [inline source/computer/linux101.rst:249]: Is this part necessary?

File: source/computer/linux101.rst [hunk 10]
```diff
@@ -49,15 +49,317 @@ Linux 入门
...
+
+    # 建立 source 目录的软链接
+    $ ln -s source source-soft
+    $ ls -l
+    total 0
+    -rw-r--r-- 2 seismo-learn seismo-learn  0 Feb  8 14:55 hello-world-hard.txt
+    -rw-r--r-- 2 seismo-learn seismo-learn  0 Feb  8 14:55 hello-world.txt
+    lrwxrwxrwx 1 seismo-learn seismo-learn 14 Feb  8 14:57 hello-world-soft.txt -> hello-world.txt
+    drwxr-xr-x 2 seismo-learn seismo-learn  6 Feb  8 14:55 source
+    lrwxrwxrwx 1 seismo-learn seismo-learn  6 Feb  8 14:58 source-soft -> source
```
Comment (seisman) [inline source/computer/linux101.rst:236]: Much shorter:
```suggestion
    # 使用 ln（link，即链接）命令建立 hello-world.txt 的硬链接 hello-world-hard.txt
    $ ln hello-world.txt hello-world-hard.txt

    # 建立 hello-world.txt 的软链接（也叫符号链接）hello-world-soft.txt
    $ ln -s hello-world.txt hello-world-soft.txt

    # 建立 source 目录的软链接
    $ ln -s source source-soft
    
    # 使用 ls 命令的 -l 选项可以查看文件和目录的详细信息
    $ ls -l
    total 0
    -rw-r--r-- 2 seismo-learn seismo-learn  0 Feb  8 14:55 hello-world-hard.txt
    -rw-r--r-- 2 seismo-learn seismo-learn  0 Feb  8 14:55 hello-world.txt
    lrwxrwxrwx 1 seismo-learn seismo-learn 14 Feb  8 14:57 hello-world-soft.txt -> hello-world.txt
    drwxr-xr-x 2 seismo-learn seismo-learn  6 Feb  8 14:55 source
    lrwxrwxrwx 1 seismo-learn seismo-learn  6 Feb  8 14:58 source-soft -> source
```

File: source/computer/linux101.rst [hunk 11]
```diff
@@ -49,15 +49,317 @@ Linux 入门
...
+    # 删除 hello-world.txt 文件的软链接
+    $ rm hello-world-soft.txt
+    $ ls
+    hello-world-hard.txt  hello-world.txt  source
+    # 删除 hello-world.txt 文件的硬链接
+    $ rm hello-world-hard.txt
+    $ ls
+    hello-world.txt  source
+
+.. note::
```
Comment (seisman) [inline source/computer/linux101.rst:251]:
```suggestion
.. admonition:: 硬链接与软链接的区别
```

File: source/computer/linux101.rst [hunk 12]
```diff
@@ -49,15 +49,317 @@ Linux 入门
...
+
+   软链接（也叫符号链接）类似 Windows 系统的快捷方式，是一个\ **文件**\ ，里面存放的
+   是源文件（或目录）的路径。删除软链接，对源文件（或目录）没有任何影响。
+   删除源文件（或目录），软链接依然存在，但无法通过其访问源文件（或目录）了。
+   软链接一般会高亮显示。
+
 文件路径
 ^^^^^^^^^
 
+访问文件或目录需要指定文件或目录的路径，Linux 下有两种表示路径的方式：绝对路径和相对路径。
```
Comment (seisman) [inline source/computer/linux101.rst:266]:
```suggestion
访问文件或目录需要指定文件或目录的路径。Linux 下有两种表示路径的方式：绝对路径和相对路径。
```

File: source/computer/linux101.rst [hunk 13]
```diff
@@ -49,15 +49,317 @@ Linux 入门
...
+       ..    Desktop          Documents
+
+   可以看出，其实家目录下存在 :file:`.` 和 :file:`..` 这两个特殊的路径。
+
 文件权限
 ^^^^^^^^
 
+Linux 下每个文件和目录都有自己的权限，使用以下命令查看文件或目录的权限::
+
+    # 进入 ~/workspace 目录，并新建 hello-world.sh 文加和 source 目录
```
Comment (seisman) [inline source/computer/linux101.rst:310]:
```suggestion
    # 进入 ~/workspace 目录，并新建 hello-world.sh 文件和 source 目录
```

File: source/computer/linux101.rst [hunk 14]
```diff
@@ -49,15 +49,317 @@ Linux 入门
...
+    $ ls -l hello-world.sh
+    -rw-rw-r-- 1 seismo-learn seismo-learn 0 Feb  7 22:37 hello-world.sh
+
+    # 增加 hello-world.sh 的可执行属性
+    $ chmod +x hello-world.sh
+    -rwxrwxr-x 1 seismo-learn seismo-learn 0 Feb  7 22:37 hello-world.sh
+
+有时候我们使用 Linux 的命令或安装程序时，可能由于没有读写某些文件的权限，而无法运行命令或
+安装程序。这时可以使用 ``sudo`` 命令临时获得 root 用户的权限。例如，在 Fedora 下安装
+GNU fortran::
```
Comment (seisman) [inline source/computer/linux101.rst:359]:
```suggestion
GNU Fortran::
```

File: source/computer/linux101.rst (hunk 9 above)
Comment (core-man) [reply source/computer/linux101.rst]: New babies may have to how to remove the link, but I simplified it in dbffffc.

**Review by seisman — APPROVED** [review]

**Pull request merged**